### PR TITLE
for production, set up CUPS config in /var for read-only root partition

### PR DESCRIPTION
**Pull request by benadida**

### CODE REVIEW

File: config/apparmor.d/usr.sbin.cups-browsed [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  #include <abstractions/base>
+  #include <abstractions/nameservice>
+  #include <abstractions/cups-client>
+  #include <abstractions/dbus>
+  #include <abstractions/p11-kit>
+
+  /{var/,}etc/cups/cups-browsed.conf r,
+  /{var/,}etc/cups/lpoptions r,
+  /{var/,}etc/cups/ppd/* r,
+  /{var/,}run/cups/certs/* r,
```
Comment (benadida) [inline config/apparmor.d/usr.sbin.cups-browsed:13]: lines changed from default, adding `{var/,}` to enable CUPS to access the new location.

File: config/apparmor.d/usr.sbin.cupsd [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  deny /dev/tty rw,  # silence noise
+  /dev/ttyS* rw,
+  /dev/ttyUSB* rw,
+  /dev/usb/lp* rw,
+  /dev/bus/usb/ r,
+  /dev/bus/usb/** rw,
+  /dev/parport* rw,
+  /{var/,}etc/cups/ rw,
+  /{var/,}etc/cups/** rw,
+  /{var/,}etc/cups/interfaces/* ixrw,
```
Comment (benadida) [inline config/apparmor.d/usr.sbin.cupsd:63]: changes from default to enable access to `/var`

File: config/apparmor.d/usr.sbin.cupsd [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  unix peer=(label=/usr/sbin/cupsd),
+
+  @{PROC}/*/auxv r,
+
+  /{usr/,}bin/dash ixr,
+  /{usr/,}bin/bash ixr,
+  /{usr/,}bin/cp ixr,
+  /etc/papersize r,
+  /{var/,}etc/cups/cups-pdf.conf r,
+  /{var/,}etc/cups/ppd/*.ppd r,
```
Comment (benadida) [inline config/apparmor.d/usr.sbin.cupsd:202]: changes from default, same.

File: config/cups-files.conf [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+#RequestRoot /var/spool/cups
+
+# Location of helper programs...
+#ServerBin /usr/lib/cups
+
+# SSL/TLS keychain for the scheduler...
+#ServerKeychain ssl
+
+# Location of other configuration files...
+ServerRoot /var/etc/cups
```
Comment (benadida) [inline config/cups-files.conf:82]: change from default

File: config/cups-files.conf [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+# Location of Samba configuration file...
+#SMBConfigFile 
+
+# Location of scheduler state files...
+StateDir /var/run/cups
```
Comment (benadida) [inline config/cups-files.conf:88]: change from default

File: config/cups.service [hunk 6]
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=CUPS Scheduler
+Documentation=man:cupsd(8)
+After=sssd.service
+
+[Service]
+ExecStart=/usr/sbin/cupsd -c /var/etc/cups/cupsd.conf -s /var/etc/cups/cups-files.conf -l
```
Comment (benadida) [inline config/cups.service:7]: change only to this line to explicitly feed new config locations.

File: config/cupsd.conf [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+LogLevel warn
+PageLogFormat
+
+# Deactivate CUPS' internal logrotating, as we provide a better one, especially
+# LogLevel debug2 gets usable now
+MaxLogSize 0
+
+# Only listen for connections from the local machine.
+Listen localhost:631
+Listen /var/run/cups/cups.sock
```
Comment (benadida) [inline config/cupsd.conf:17]: single change here

**Review by umbernhard — APPROVED** [review]

**Pull request merged**